AllReadyApp/Web-App/AllReady.UnitTest/Services/Routing/GoogleOptimizeRouteServiceShould.cs

```diff
@@ -7,33 +7,129 @@
+                mockedHttpClient.Object);
+    
+            var result = await service.OptimizeRoute(_optimiseRouteCriteria);
+
+            mockedHttpClient.Verify(x => x.GetAsync(It.Is<string>(y => y == expectedApiCall)));
+        }
+
+        [Fact]
+        public async Task ReturnsNull_WhenOnBadAPIRequest()
+        {
+            var mappingSettings = new MappingSettings() { GoogleDirectionsApiKey = "some key" };
+            var options = Options.Create(mappingSettings);
+            var mockedHttpClient = new Mock<IHttpClient>();
+            mockedHttpClient.Setup(x => x.GetAsync(It.IsAny<string>()))
+                .Returns(Task.FromResult(new HttpResponseMessage(System.Net.HttpStatusCode.BadRequest)));
+            var service = new GoogleOptimizeRouteService(
+                options,
+                Mock.Of<ILogger<GoogleOptimizeRouteService>>(),
+                mockedHttpClient.Object);
+
+            var result = await service.OptimizeRoute(_optimiseRouteCriteria);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task ReturnNull_WhenApiCallTimesOut()
+        {
+            var mappingSettings = new MappingSettings() { GoogleDirectionsApiKey = "some key" };
+            var options = Options.Create(mappingSettings);
+            var mockedHttpClient = new Mock<IHttpClient>();
+            mockedHttpClient.Setup(x => x.GetAsync(It.IsAny<string>())).Throws(new TaskCanceledException());
+            var service = new GoogleOptimizeRouteService(
+                options,
+                Mock.Of<ILogger<GoogleOptimizeRouteService>>(),
+                mockedHttpClient.Object);
+
+            var result = await service.OptimizeRoute(_optimiseRouteCriteria);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task ReturnNull_WhenGoogleDirectionsAPIKeyIsEmpty()
         {
             var mappingSettings = new MappingSettings() { GoogleDirectionsApiKey = string.Empty };
             var options = Options.Create(mappingSettings);
-            var mockedLogger = new Mock<ILogger<GoogleOptimizeRouteService>>();
             var mockedHttpClient = new Mock<IHttpClient>();
             var service = new GoogleOptimizeRouteService(
                 options,
-                mockedLogger.Object,
+                Mock.Of<ILogger<GoogleOptimizeRouteService>>(),
                 mockedHttpClient.Object);
 
-            var optimiseRouteCriteria = new OptimizeRouteCriteria(
-                "some starting address",
-                "test end address",
-                new List<OptimizeRouteWaypoint>()
-                    {
-                        new OptimizeRouteWaypoint(100.0, 100.0, new Guid())
-                    }
-                );
+            var result = await service.OptimizeRoute(_optimiseRouteCriteria);
 
-            var result = await service.OptimizeRoute(optimiseRouteCriteria);
             Assert.Null(result);
         }
     }
```

Comment:
@chinwobble if you search the codebase for other classes using ILogger<T>, and find the unit tests for those classes, there are examples of how to mock the ILogger to assert the correct thing is logged.